src/Core/Components/Slider/FluentSlider.razor.cs

```diff
@@ -0,0 +1,124 @@
+    /// </summary>
+    [Parameter]
+    public RenderFragment? ChildContent { get; set; }
+
+    /// <summary>
+    /// Gets or sets the value of the slider.
+    /// </summary>
+    protected override void OnParametersSet()
+    {
+        base.OnParametersSet();
+
+        if (typeof(TValue) != typeof(byte) &&
+           typeof(TValue) != typeof(int) &&
+           typeof(TValue) != typeof(long) &&
+           typeof(TValue) != typeof(float) &&
+           typeof(TValue) != typeof(double) &&
+           typeof(TValue) != typeof(decimal))
+        {
+            throw new InvalidOperationException("FluentSlider only supports numeric types.");
+        }
+    }
+
+    /// <summary>
+    /// Formats the value as a string. Derived classes can override this to determine the formatting used for <c>CurrentValueAsString</c>.
+    /// </summary>
+    /// <param name = "value">The value to format.</param>
+    /// <returns>A string representation of the value.</returns>
+    protected override string? FormatValueAsString(TValue? value)
+    {
+        // Avoiding a cast to IFormattable to avoid boxing.
+        return value switch
+        {
+            null => null,
+            int @int => BindConverter.FormatValue(@int, CultureInfo.InvariantCulture),
+            long @long => BindConverter.FormatValue(@long, CultureInfo.InvariantCulture),
+            float @float => BindConverter.FormatValue(@float, CultureInfo.InvariantCulture),
+            double @double => BindConverter.FormatValue(@double, CultureInfo.InvariantCulture),
+            decimal @decimal => BindConverter.FormatValue(@decimal, CultureInfo.InvariantCulture),
+            _ => throw new InvalidOperationException($"Unsupported type {value.GetType()}"),
+        };
+    }
+
+    /// <summary>
+    /// Handler for the OnFocus event.
+    /// </summary>
+    /// <param name="e"></param>
+    /// <returns></returns>
+    protected virtual Task FocusOutHandlerAsync(FocusEventArgs e)
+    {
+        FocusLost = true;
+        return Task.CompletedTask;
+    }
+
+    /// <summary>
+    /// Parses a string to create the <see cref="Microsoft.AspNetCore.Components.Forms.InputBase{TValue}.Value"/>.
+    /// </summary>
+    /// <param name="value">The string value to be parsed.</param>
+    /// <param name="result">The result to inject into the Value.</param>
+    /// <param name="validationErrorMessage">If the value could not be parsed, provides a validation error message.</param>
+    /// <returns>True if the value could be parsed; otherwise false.</returns>
+    protected override bool TryParseValueFromString(string? value, [MaybeNullWhen(false)] out TValue result, [NotNullWhen(false)] out string? validationErrorMessage)
+    {
+        return this.TryParseSelectableValueFromString(value, out result, out validationErrorMessage);
+    }
+
+    // Only for Unit Tests
+    internal string? ValidateFormatValueAsString(TValue? value) => FormatValueAsString(value);
```

Comment:
You can replace this method with this one to be used in the razor file when a null value can be returned.
```csharp
internal string? FormatValueAsStringOrNull(TValue? value)
{
    if (value is null)
    {
        return null;
    }

    return FormatValueAsString(value.Value);
}
```